File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/interfaces/YGLogger.java
package io.github.orioncraftmc.meditate.internal.interfaces;

import io.github.orioncraftmc.meditate.internal.YGConfig;
import io.github.orioncraftmc.meditate.internal.YGNode;
import io.github.orioncraftmc.meditate.internal.enums.YGLogLevel;

@FunctionalInterface
public interface YGLogger {
    int invoke(YGConfig config, YGNode node, YGLogLevel level, String format, Object... args);
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/internal/interfaces/YGLogger.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga.internal.interfaces

import app.cash.redwood.yoga.internal.YGConfig
import app.cash.redwood.yoga.internal.YGNode
import app.cash.redwood.yoga.internal.enums.YGLogLevel

internal fun interface YGLogger {
  operator fun invoke(
    config: YGConfig?,
    node: YGNode?,
    level: YGLogLevel,
    format: String,
    vararg args: Any?,
  ): Int
}
